lib/readMaidFile.js

```diff
@@ -1,13 +1,44 @@
+const fs = require('fs')
 const parseMarkdown = require('./parseMarkdown')
-const loadFile = require('./loadFile')
+const find = require('find-file-up')
 
-module.exports = ({ section } = {}) => {
-  const { path: filepath, data } = loadFile.loadSync([
-    'maidfile.md',
-    'README.md',
-    'readme.md'
-  ])
-  if (!filepath) return null
+// a bit duplication with what we have in parseMarkdown.js, but we can't DRY it
+const MAIDFILE = 'maidfile.md'
 
-  return parseMarkdown(data, { section, filepath })
+/**
+ * Resolving mechanism. Always starts to search for `maidfile.md` from
+ * the current working directory to 5 levels upwards (probably enough).
+ * Then repeats this for `.maidfile.md` (with dot) but 10 levels upwards,
+ * because who knows, someone may have big folder trees.
+ *
+ * Second step. If it can't find the Maid files (above ones) and there is
+ * no given `--config-path / -c` flag on the CLI it throws with error message
+ * thrown from the main Maid class constructor.
+ *
+ * Third step. If config path is given, it treats that config file
+ * a bit more specifically, as described in the (current) readme - task names should be
+ * h3 headers and should have some h2 header, which in turn can also be defined on
+ * the CLI through the `--section` flag.
+ *
+ * @param {Object} opts command line global flags
+ * @returns {Object} like `{ filepath, tasks }`, coming from parseMarkdown
+ */
+module.exports = (opts = {}) => {
+  let filepath =
+    find.sync(MAIDFILE, opts.cwd, 5) || find.sync(`.${MAIDFILE}`, opts.cwd, 10)
```

Comment:
oh.. i forgot that, but it's not needed for `find-file-up` to work it uses the cwd by default anyway.

Reply:
use of the `--cwd` flag is not encouraged anyway, because it also may conflict with other CLIs (#35)